packages/core/src/State.ts

```diff
@@ -47,21 +47,21 @@ export function stateValuesEqual(
   );
 }
 
-export function isState<
-  TContext,
-  TEvent extends EventObject,
-  TStateSchema extends StateSchema<TContext> = any,
-  TTypestate extends Typestate<TContext> = { value: any; context: TContext }
->(
-  state: object | string
-): state is State<TContext, TEvent, TStateSchema, TTypestate> {
-  if (isString(state)) {
+export function isStateConfig<TContext, TEvent extends EventObject>(
+  state: any
+): state is StateConfig<TContext, TEvent> {
+  if (typeof state !== 'object' || state === null) {
     return false;
   }
 
-  return 'value' in state && 'history' in state;
+  return 'value' in state && '_sessionid' in state;
```

Comment:
`_sessionid` is not essential - cause it won't really be reused/used anyhow (IIRC). I see how this helps to check if something is a persisted state or not but maybe it would be better to check for something more "required". 

It's kinda a nit level so it doesn't matter.

Reply:
`_event` perhaps?

Reply:
Works for me.